### ISSUE TITLE
Possibility to correlate http requests custom headers with their respective traces

### ISSUE THREAD

**Issue by rodrigo** [newrelic/newrelic-flutter-agent#127]
I'm creating a flutter integration for my mobile app that uses GraphQL to make requests to our backend so as you can imagine, every request/trace just have the same endpoint/method. I have added the `addHTTPHeadersTrackingFor` to track the operationName of each query/mutation. I was hoping to select the requests in the "HTTP requests" and use the "view distributed traces" to filter them, but when I select the request using the operationName grouping, the option to view distributed traces just don't appear.
The http request screen if I'm filtering by request domain:
![image](https://github.com/user-attachments/assets/2fcf226e-1d4d-4168-b6e1-d37f5eddfa93)
The same screen if I'm filtering by operation name:
![image](https://github.com/user-attachments/assets/b4e1780c-78b4-4456-98fa-e7801eb5dc20)
I'm I missing some configuration? Other thing that would help me would be the possibility to change the trace name by itself, but I didn't find any mention to this on the docs.

**Comment by ndesai-newrelic** [comment]
@rodrigo Can you share which GraphQL Library you are using? Also, can you share a code snippet for a GraphQL request? This will help us reproduce the issue and provide a solution.

**Comment by rodrigo** [comment]
@ndesai-newrelic
We're using the [graphql_flutter lib on the version 5.1.0](https://pub.dev/packages/graphql_flutter/versions/5.1.0#installation).
On our main.dart, we added the tracking method:
```
    NewrelicMobile.instance.addHTTPHeadersTrackingFor(['operationName']);
```

For the graphql requests we have a class (still in development) that the implements [HttpLink](https://pub.dev/documentation/gql_http_link/latest/gql_http_link/HttpLink-class.html) and change the request method to add the header:
```
class CustomLink implements HttpLink {
...
  @override
  Stream<Response> request(Request request, [NextLink? forward]) =>
      _delegate.request(
          request.updateContextEntry<HttpLinkHeaders>(
            (HttpLinkHeaders? headers) => HttpLinkHeaders(
              headers: <String, String>{
                ...headers?.headers ?? <String, String>{},
                'operationName':
                    exp.firstMatch(request.operation.toString())![0] == null
                        ? 'Anonymous Query'
                        : exp
                            .firstMatch(request.operation.toString())![0]
                            .toString()
              },
            ),
          ),
          forward);
```
A class that builds a `GraphQLClient` using this custom the `CustomLink` when the app launches. And finally parts of the code that use the client to make requests:
```
  Future<Account> requestAccount() async {
    final resultData = await doGraphQlOperation(
      () => client.query(QueryOptions(
        document: gql(someQuery),
        fetchPolicy: FetchPolicy.noCache,
      )),
    );
...
```

**Comment by ndesai-newrelic** [comment]
@rodrigo Is the issue you described happening on Android or iOS? I found a bug in iOS where we are not tracking headers, and I am going to fix it in the next release.

**Comment by ndesai-newrelic** [comment]
https://github.com/newrelic/newrelic-flutter-agent/pull/128

**Comment by rodrigo** [comment]
Hi @ndesai-newrelic 
The issue is happening on both android and ios!

**Comment by rodrigo** [comment]
@ndesai-newrelic I don't know the outcome of the bug fixed on #128, but unfortunately I don't think its the same case. The headers are being tracked, I can see them in the selection that groups the http request. The problem is that when I select an specific http request through a custom header filter, I can't view the related traces.

**Comment by ndesai-newrelic** [comment]
@rodrigo let's discuss this on newrelic forum. I am closing it here.
